File: src/main/java/com/employee/product/employeedetails/response/dto/EmployeeDetailsResponseDto.java
package com.employee.product.employeedetails.response.dto;


import java.util.List;

import lombok.Data;

@Data
public class EmployeeDetailsResponseDto {

	private String id;

	private String firstName;

	private String lastName;

	private String emailId;

	private String sex;

	private String addressLine1;

	private String addressLine2;

	private String city;

	private String state;

	private String country;

	private String contactNumber;

	private String dateOfBirth;

	private String postalCode;

	private String jobRole;

	private String department;

	private String reportingPerson;

	private String workLocation;

	private List<EmployeeWorkPermitDetailsResponseDto> workPermitDetails;

	private List<EmployeePassportDetailsResponseDto> passportDetails;

	private List<EmployeeFamilyDetailsResponseDto> familyDetails;

	// Commenting the call to map Payslip details as it is not required in this
	// module
	// private List<EmployeePaySlipDocumentDetailsResponseDto> payslipDetails;

}

File: src/main/js/eweb/src/app/ehome/employeeDetailsResponseDto.ts
import {EmployeeFamilyDetailsResponseDto} from './employeeFamilyDetailsResponseDto';
import {EmployeePassportDetailResponseDto} from './employeePassportDetailResponseDto';
import {EmployeeWorkPermitDetailsResponseDto} from './employeeWorkPermitDetailsResponseDto';

export class EmployeeDetailsResponseDto {
id: string;
firstName: string;
lastName: string;
emailId: string;
sex: string;
addressLine1: string;
addressLine2: string;
city: string;
state: string;
country:string;
contactNumber: string;
dateOfBirth: string;
reportingPerson: string;
jobRole:string;
workLocation:string;
department:string;
employeeWorkPermitDetailsResponseDto:EmployeeWorkPermitDetailsResponseDto[];
employeePassportDetailResponseDto:EmployeePassportDetailResponseDto[];
employeeFamilyDetailsResponseDto:EmployeeFamilyDetailsResponseDto[];

constructor(){}
}
